extension/httpfs/include/crypto.hpp

```diff
@@ -16,4 +17,42 @@ void hmac256(std::string message, hash_bytes secret, hash_bytes &out);
 
 void hex256(hash_bytes &in, hash_str &out);
 
+class AESGCMStateSSL : public duckdb::AESGCMState {
+
+public:
+	DUCKDB_API explicit AESGCMStateSSL();
+	DUCKDB_API ~AESGCMStateSSL() override;
+
+public:
+	DUCKDB_API bool IsOpenSSL() override;
+	DUCKDB_API void InitializeEncryption(duckdb::const_data_ptr_t iv, duckdb::idx_t iv_len,
```

Comment:
nit: since this is in the duckdb namespace the `duckdb::` identifiers are unnecessary